pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -339,6 +339,7 @@ func (c *Cluster) provisionPodTemplateSpec(devices []rookalpha.Device, selection
 		RestartPolicy: restart,
 		Volumes:       volumes,
 		HostNetwork:   c.HostNetwork,
+		HostIPC:       true,
```

Comment:
A couple questions:
- Could we only set HostIPC if devices are being specified and if the encryption is being specified?
- Did you say you also needed a mount for `/var/run/udev`? If so, we could add it in the same condition as setting the HostIPC flag. If you add it, don't forget both the volume and volumeMount. Too frequently I miss one of them and then wonder why it fails...

Reply:
Yes to the first, I'll fix that up soon. I needed /var/run/udev to create udev cookies manually with dmsetup in a different set of tests and thought that'd carry over, but it looks like the cryptsetup library does this without accessing the file system.

Reply:
It makes sense to have it more granular (only enable HostIPC for encryption), however, I have the impression that this might be required by other calls in the future. So leaving it as is (enabled by default for everything) might prevent us from falling into other bugs.

Reply:
granular now